### PR TITLE
smallab 0.1.0

### PR DESCRIPTION
**Pull request by octopuscabbage**

### CODE REVIEW

**Comment by octopuscabbage** [comment]
solve issue #6 and issue #9, Implements new folder structure

File: demo.py [hunk 1]
```diff
@@ -18,6 +18,8 @@ def main(self, specification: typing.Dict) -> typing.Dict:
             random.seed(specification["seed"])
             for i in range(specification["num_calls"]): #Advance the random number generator some amount
                random.random()
+            if "fail" in specification and specification["fail"]:
```
Comment (akumaraguru) [inline demo.py]: What's the point of this `fail` parameter?

File: demo.py [hunk 2]
```diff
@@ -59,13 +62,13 @@ def main(self, specification: typing.Dict) -> typing.Dict:
...
 
     #Read back our results
-    for fname in os.listdir(runner.get_save_directory("random_number_from_generator")):
-        if "json" not in fname: #don't read back the completed file
-            with open(os.path.join(runner.get_save_directory("random_number_from_generator"), fname), "rb") as f:
-                results = pickle.load(f)
-                print(results["specification"]["seed"])
-                print(results["result"]["number"])
-
+    for root,_,files in os.walk(runner.get_save_directory("random_number_from_generator")):
```
Comment (akumaraguru) [inline demo.py]: For the future, maybe add helper methods to do the directory walk.

File: demo.py [hunk 3]
```diff
@@ -90,3 +93,13 @@ def main(self, specification: typing.Dict) -> typing.Dict:
     #That together make the whole specification
     assert specifications_1.union(specifications_2) == all_specifications
 
+
+
+    #You can use the provided logging callbacks to log completion and failure of specific specifcations
+    from smallab.utilities import logger_callbacks
+    name = "with_logging"
```
Comment (akumaraguru) [inline demo.py]: For the future, move the callback logic out of the runner.

File: smallab/runner.py [hunk 4]
```diff
@@ -168,24 +201,28 @@ def parallel_f(name,experiment,specification):
 
 
     def __run_and_save(self, name, experiment, specification,dont_catch_exceptions):
+        def _interior_fn():
```
Comment (akumaraguru) [inline smallab/runner.py]: Maybe a more descriptive name?

**Pull request merged**